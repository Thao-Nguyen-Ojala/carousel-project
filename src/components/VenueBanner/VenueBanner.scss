$wrapper-height: 13rem;
$text-height: $wrapper-height / 2;
$statusHeight: 18rem;
$statusLineHeight: $statusHeight / 2;
//media max-width 959px
$venueWidthBanner959px: 19rem;
$venueWidthBanner500px: 22rem;

.venue-banner {
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 22rem;
  height: 20rem;
  margin: auto;

  border: 1px solid #E4E4E4;
  border-radius: 10px;
  box-shadow: 2px 2px 10px #E4E4E4;

  &__img-section {
    position: absolute;
    display: block;
    justify-content: center;

    &__img {
      opacity: 1;
      width: 22rem !important;
      height: 15rem !important;
      vertical-align: middle;
     
    }
  }

  &__restaurant-name-section {
    color: #2e2f35;
    font-size: 20px;
    text-align: center;
  }
 
  &__status {
    width: 22rem;
    height: $statusHeight;
    text-align: center;
    font-size: 20px;
    font-weight: bold;
    position: absolute;
    line-height: $text-height + $statusLineHeight;
  }

  &__offline {
    position: relative;
    opacity: 0.2;
  }

  &__online {
    position: relative;
    opacity: 1;
  }

  &__status-light-text {
    color: #646572;
  }

  &:hover {
    cursor: pointer;
    transform: scale(1.02);
  }
}

@media screen and (max-width: 959px) {
  .venue-banner {
    width: $venueWidthBanner959px;
    height: 18rem;
    margin: auto;

    &__img-section {
      &__img {
        width: $venueWidthBanner959px !important;
        height: 14rem !important;
      }
    }

    &__status {
      width: $venueWidthBanner959px;
      line-height: 14rem;
    } 
    
  }
}


