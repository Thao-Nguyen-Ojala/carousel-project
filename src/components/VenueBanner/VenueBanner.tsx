import { Blurhash } from 'react-blurhash';
import './VenueBanner.scss';

type VenueBannerPropsType = {
  blurhash: string;
  name: string;
  online: boolean;
};

const VenueBanner = ({ blurhash, name, online }: VenueBannerPropsType) => {
  const statusClass = !online
    ? 'venue-banner__offline'
    : 'venue-banner__online';
  const blurhashClasses = 'venue-banner__img-section__img ' + statusClass;
  return (
    <div className="venue-banner">
      <div className="venue-banner__img-section">
        <Blurhash
          className={blurhashClasses}
          hash={blurhash}
          resolutionX={32}
          resolutionY={32}
          punch={1}
        />

        <div className="venue-banner__restaurant-name-section">{name}</div>
      </div>
      <div className="venue-banner__status">
        {online ? (
          <span>{'Online'} </span>
        ) : (
          <span className="venue-banner__status-light-text">{'Closed'} </span>
        )}
      </div>
    </div>
  );
};

export default VenueBanner;
