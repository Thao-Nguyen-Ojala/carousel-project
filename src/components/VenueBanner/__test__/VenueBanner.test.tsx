import React from 'react';
import ReactDOM from 'react-dom';
import VenueBanner from '../VenueBanner';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-canvas-mock';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('matches VenueBanner snapshot', () => {
  const testingVenueBanner = renderer
    .create(
      <VenueBanner
        blurhash={'UAN=8k?LS~M:ErJFs%t0MDMWRqo@%BxSV{RX'}
        name={'Sea Chain'}
        online={true}
      ></VenueBanner>
    )
    .toJSON();
  expect(testingVenueBanner).toMatchSnapshot();
});
