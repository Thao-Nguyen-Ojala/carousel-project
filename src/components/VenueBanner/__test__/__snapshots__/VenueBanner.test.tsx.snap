// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches VenueBanner snapshot 1`] = `
<div
  className="venue-banner"
>
  <div
    className="venue-banner__img-section"
  >
    <div
      className="venue-banner__img-section__img venue-banner__online"
      style={
        Object {
          "display": "inline-block",
          "height": 128,
          "position": "relative",
          "width": 128,
        }
      }
    >
      <canvas
        height={32}
        punch={1}
        style={
          Object {
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
        width={32}
      />
    </div>
    <div
      className="venue-banner__restaurant-name-section"
    >
      Sea Chain
    </div>
  </div>
  <div
    className="venue-banner__status"
  >
    <span>
      Online
       
    </span>
  </div>
</div>
`;
